Source/JavaScriptCore/runtime/Options.cpp

```diff
@@ -635,8 +659,8 @@ static inline void disableAllJITOptions()
     Options::useJITCage() = false;
     Options::useConcurrentJIT() = false;
 
-    if (!Options::useInterpretedJSEntryWrappers())
-        Options::useWasm() = false;
+    if (!Options::useInterpretedJSEntryWrappers() && Options::useWasm())
```

Comment:
Why do you need the `Options::useWasm()` condition? Seems like we could just disable all wasm options when we can't run wasm? Also, it seems like setting `Options::useWasm() = false` would trigger `notifyOptionsChanged()` to disable all wasm options anyway so I don't think any change was needed here.